### ISSUE TITLE
FN useless_format: fmt::DIsplay impl

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#6773]
Lint name: useless_format

I tried this code:

```rust
use std::fmt;

struct S {}

impl fmt::Display for S {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "")
    }
}

fn main() {
    let x: String = format!("{}", S {});
    // could just be:
    // let x = S{}.to_string();
}
```
IMO clippy should issue a `useless_format` warning here, I'm not sure why it does not.

### Meta

- `cargo clippy -V`: `clippy 0.1.52 (07194ff 2021-02-10)`
````
rustc 1.52.0-nightly (07194ffcd 2021-02-10)
binary: rustc
commit-hash: 07194ffcd25b0871ce560b9f702e52db27ac9f77
commit-date: 2021-02-10
host: x86_64-unknown-linux-gnu
release: 1.52.0-nightly
LLVM version: 11.0.1
````

**Comment by camsteffen** [comment]
Duplicate of #3753
